octoprint_octorelay/__init__.py

```diff
@@ -178,14 +178,16 @@ def on_event(self, event, payload):
         #elif event == Events.PRINT_CANCELLED:
             # self.print_stopped()
 
-    def handle_plugin_event(self, event):
-        self._logger.debug(f"Handling the plugin event: {event}")
+    def handle_plugin_event(self, event, scope = RELAY_INDEXES):
+        self._logger.debug(f"Handling the plugin event {event} having scope: {scope}")
         settings = self._settings.get([], merged=True) # expensive
-        for index in RELAY_INDEXES:
+        for index in scope:
             if bool(settings[index]["active"]):
                 target = settings[index]["rules"][event]["state"]
                 if target is not None:
                     target = bool(target)
+                    if target and event == TURNED_ON:
+                        continue # avoid infinite loop
```

Comment:
```suggestion
                    if target and event == TURNED_ON:
                        self._logger.debug(f"Skipping {index} to avoid infinite loop")
                        continue # avoid infinite loop
```